ecs/basicsystem.go

```diff
@@ -0,0 +1,79 @@
+package ecs
+
+import "runtime"
+
+// BasicSystemWrapper is the default implementation of the System interface.
+type BasicSystemWrapper struct {
```

Comment:
Well, the name doesn't really matter to me (perhaps @paked has a preference). But it *is* necessary: if I define these functions in some kind of struct, then there is no way to override the `Pre()`, `EUpdate()` and `Post()` functions.